src/pages/groupComparison/GroupComparisonStore.ts

```diff
@@ -0,0 +1,84 @@
+import {ResultsViewPageStore} from "../resultsView/ResultsViewPageStore";
+import {remoteData} from "../../shared/api/remoteData";
+import ListIndexedMap from "../../shared/lib/ListIndexedMap";
+import {Sample, SampleIdentifier} from "../../shared/api/generated/CBioPortalAPI";
+import { computed } from "mobx";
+import { PatientSurvival } from "shared/model/PatientSurvival";
+import * as _ from "lodash";
+import { COLORS } from "pages/studyView/StudyViewUtils";
+
+export default class GroupComparisonStore {
+    constructor(
+        private resultsViewPageStore:ResultsViewPageStore
+    ) {
+    }
+
+    public readonly sampleGroups = remoteData({
+        await:()=>[
+            this.resultsViewPageStore.samples,
+            this.resultsViewPageStore.sampleGroups
+        ],
+        invoke:()=>{
+            const sampleIdentifierToSample = new ListIndexedMap<Sample>();
+            for (const sample of this.resultsViewPageStore.samples.result!) {
+                sampleIdentifierToSample.set(sample, sample.sampleId, sample.studyId);
+            }
+
+            let colorIndex = 0;
+            return Promise.resolve(
+                this.resultsViewPageStore.sampleGroups.result!.map((group)=>({
```

Comment:
i think we decided that resultsViewPageStore should not maintain sampleGroups collection.  These groups should be in the groupComparison store